opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -113,6 +138,17 @@
          */
         SpanBuilder addReference(String referenceType, SpanContext referencedContext);
 
+        /**
+         * Remove any explicit (e.g., via {@link SpanBuilder#addReference(String,SpanContext)}) or implicit (e.g., via
+         * {@link ActiveSpanSource#activeContext()}) references to parent / predecessor SpanContexts, thus making the built
+         * Span a "root" of a Trace tree/graph.
```

Comment:
I don't agree that is should _remove_ anything that's explicitly added to the builder. The typical builder pattern is agnostic to the order of set/with method calls, it just accumulates the state and uses it in the `build()`. So it shouldn't matter if references are added before or after `asRoot()` is called.

I think it should only break the implicit reference to the active span.